src/Datadog.Trace/Agent/Api.cs

```diff
@@ -57,16 +58,58 @@ public async Task SendTracesAsync(IList<List<Span>> traces)
 
         private async Task SendAsync<T>(T value, Uri endpoint)
         {
+            MsgPackContent<T> content;
             try
             {
-                var content = new MsgPackContent<T>(value, _serializationContext);
-                var response = await _client.PostAsync(endpoint, content);
-                response.EnsureSuccessStatusCode();
+                content = new MsgPackContent<T>(value, _serializationContext);
             }
             catch (Exception ex)
             {
-                _log.ErrorException("An error occured while sending traces to the agent at {Endpoint}", ex, endpoint);
+                _log.ErrorException("An error occured while sending serializing traces", ex);
+                return;
             }
+
+            // retry up to 5 times with exponential backoff
+            var retryLimit = 5;
+            var retryCount = 1;
+            var sleepDuration = 100; // in milliseconds
+
+            while (true)
+            {
+                try
+                {
+                    var response = await _client.PostAsync(endpoint, content);
+                    response.EnsureSuccessStatusCode();
+                    return;
+                }
+                catch (Exception ex)
+                {
+                    if (retryCount >= retryLimit)
+                    {
+                        _log.ErrorException("An error occured while sending traces to the agent at {Endpoint}", ex, endpoint);
+                        return;
+                    }
+                }
+
+                await Sleep(sleepDuration);
```

Comment:
I think you can use [`await Task.Delay()`](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.delay) here instead of implementing you own asynchronous `Sleep()`.

```suggestion
                await Task.Delay(sleepDuration);
```